packages/next/client/image.tsx

```diff
@@ -35,6 +35,8 @@ type ImageProps = Omit<
 
 const imageData: ImageData = process.env.__NEXT_IMAGE_OPTS as any
 const { sizes: configSizes, loader: configLoader, path: configPath } = imageData
+configSizes.sort((a, b) => a - b) // smallest to largest
+const largestSize = configSizes[configSizes.length - 1]
```

Comment:
It would be nice to be able to select a value that approximates the value of the width attribute.

Reply:
I implemented a solution in 20a5044, but this only works client side so we can't ship that because of inconsistencies with SSR.

The largest size here is predicable and safest. Remember the largest image is used as a fallback, the `srcset` feature is already approximating the value.

See browser compatibility here: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/srcset#Browser_compatibility